sql/core/src/main/scala/org/apache/spark/sql/execution/r/ArrowRRunner.scala

```diff
@@ -55,13 +56,32 @@ class ArrowRRunner(
     numPartitions = -1,
     isDataFrame = true,
     schema.fieldNames,
-    RRunnerModes.DATAFRAME_GAPPLY) {
+    mode) {
+
+  // TODO: it needs to refactor to share the same code with RRunner, and have separate
+  // ArrowRRunners.
+  private val getNextBatch = {
```

Comment:
I plan to clean this whole ArrowRRunner to deduplicate codes with RRunner after this PR.

Reply:
I filed a JIRA https://issues.apache.org/jira/browse/SPARK-26923